Lib/test/test_ipaddress.py

```diff
@@ -1494,6 +1504,12 @@ def testHosts(self):
                  ipaddress.IPv4Address('2.0.0.1')]
         str_args = '2.0.0.0/31'
         tpl_args = ('2.0.0.0', 31)
+        self.assertIsInstance(ipaddress.ip_network(str_args).hosts(), Iterator)
+        self.assertIsInstance(ipaddress.ip_network(tpl_args).hosts(), Iterator)
+        hosts = ipaddress.ip_network(str_args).hosts()
+        self.assertIsNotNone(next(hosts))
+        hosts = ipaddress.ip_network(tpl_args).hosts()
+        self.assertIsNotNone(next(hosts))
```

Comment:
You can reuse `hosts` for isinstace checks.

Also, you can test that `next(hosts)` is equal to `addrs[0]`.